ignite/cmd/plugin.go

```diff
@@ -360,6 +364,130 @@ func NewPluginUpdate() *cobra.Command {
 	}
 }
 
+func NewPluginAdd() *cobra.Command {
+	cmdPluginAdd := &cobra.Command{
+		Use:  "add [path]",
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("must provide a plugin path")
+			}
+			chain, err := newChainWithHomeFlags(cmd)
+			s := cliui.New(cliui.WithStdout(os.Stdout))
+
+			if err != nil {
+				return err
+			}
+			nodePath := chain.ConfigPath()
+			nodePath = path.Dir(nodePath)
+
+			confPath, err := pluginsconfig.LocateDefault(nodePath)
+			if err != nil {
+				return err
+			}
+			var conf pluginsconfig.Config
+			f, err := os.Open(confPath)
+			if err != nil {
+				return err
+			}
+			err = conf.Decode(f)
+			if err != nil {
+				return err
+			}
+
+			for _, p := range conf.Plugins {
+				if p.Path == args[0] {
```

Comment:
What if I add a plugin with a different version ?

Let's say I have a plugin configured as :

```yaml
plugins:
- path: github.com/org/plugin@v1
```

And I call `ignite plugin add github.com/org/plugin@v2` ?

v2 will replace v1 but since the source has already been fetched in `.ignite/plugins`, I wonder if this will be handled correctly.

Reply:
ill have to familiarize myself with how the these modules are cached inside this directory. can we parse after the @ to create a new subdirectory so in this example 
```
.ignite
-- plugins
----- plugin
```

```
.ignite
-- plugins
----- plugin
------- v2
```
this will allow for unique plugin version to be scoped in the same parent directory while also allowing explicit versioning to be respected. this can be added to the `Plugin` definition as an version property to allow for easier checks on the plugin version after path parsing from the config. right now I think it is dropped. The new `Plugin` struct would then look like: 
```
type Plugin struct {
	// Embed the plugin configuration
	pluginsconfig.Plugin
	// Interface allows to communicate with the plugin via net/rpc.
	Interface Interface
	// If any error occurred during the plugin load, it's stored here
	Error error

	repoPath   string
	cloneURL   string
	cloneDir   string
	reference  string
	srcPath    string
	binaryName string
        version     string
	client *hplugin.Client
}
```
This would then allow for an addition check on the version in this condition
```
for _, p := range conf.Plugins {
if p.Path == args[0]  && p.Version == version {
```

I think with the current implementation the paths within the config will still hold the version within its `path` property so I'm unsure on how necessary the above solution is at this time. but it would allow for multiple plugin version to be loaded without conflict.

Reply:
This is definitively the way to go, we had in mind something similar for the *global* vs *local/chain* plugin, a kind of new subdirectory after `.ignite/plugins`, again to avoid conflicts with versions. But as you suggest, using the plugin version as a subdirectory is probably the best.

That said, it's out of scope of this PR, so let's discuss that in an other issue (I'll create it later). I'll resolve this comment for now.